src/Sdk/ExceptionFactory.php

```diff
@@ -0,0 +1,58 @@
+<?php
+declare(strict_types=1);
+
+namespace LoyaltyCorp\SdkBlueprint\Sdk;
+
+use EoneoPay\Externals\HttpClient\Exceptions\InvalidApiResponseException;
+use LoyaltyCorp\SdkBlueprint\Sdk\Exceptions\CriticalException;
+use LoyaltyCorp\SdkBlueprint\Sdk\Exceptions\NotFoundException;
+use LoyaltyCorp\SdkBlueprint\Sdk\Exceptions\RuntimeException;
+use LoyaltyCorp\SdkBlueprint\Sdk\Exceptions\ValidationException;
+
+class ExceptionFactory
+{
+    /**
+     * The error code.
+     *
+     * @var \EoneoPay\Externals\HttpClient\Exceptions\InvalidApiResponseException
+     */
+    private $exception;
+
+    /**
+     * Initialize the attribute.
+     *
+     * @param \EoneoPay\Externals\HttpClient\Exceptions\InvalidApiResponseException $exception
+     */
+    public function __construct(InvalidApiResponseException $exception)
+    {
+        $this->exception = $exception;
+    }
+
+    /**
+     * Create exception object based on error code range.
+     *
+     * @return \Exception
+     */
+    public function create(): \Exception
+    {
+        $content = \json_decode($this->exception->getResponse()->getContent(), true);
+
+        $code = $content['code'] ?? $this->exception->getCode();
+
+        $message = $content['message'] ?? $content['exception'] ?? '';
+
```

Comment:
|`line 34`|
|---|
I think here we should say that it returns a base exception from utils because this way to autocompletion will be better.